### PR TITLE
Add v2 Migration Guide

### PR DESCRIPTION
**Pull request by cmarfia**
# CHANGELOG

## Added
- Migration guide for **v1** to **v2**

### CODE REVIEW

**Comment by cmarfia** [comment]
I Approve!

File: MIGRATE_TO_V2.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+Change
+```go
+// x.x is used to denote any minor or patch version for v1
+github.com/MonsantoCo/mocka v1.x.x
+```
+
+To
+```
+github.com/MonsantoCo/mocka v2.0.0
```
Comment (poost-frey) [inline MIGRATE_TO_V2.md]:
```suggestion
github.com/MonsantoCo/mocka/v2 v2.0.0
```

File: MIGRATE_TO_V2.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    t.Errorf("expected nil but got %v", err.Error())
+}
+defer stub.Restore()
+
+err = stub.Return(2)
+if err != nil {
+    t.Errorf("expected nil but got %v", err.Error())
+}
+
+err = stub.WithArgs("hello").Return(3)
```
Comment (poost-frey) [inline MIGRATE_TO_V2.md:106]: could inline the err with the `if` since they aren't returning anything else

Comment (cmarfia) [reply MIGRATE_TO_V2.md:106]: Since this is just code style in a README.md, I don't see a need for this change.

**Review by poost-frey — APPROVED** [review]

**Pull request merged**